src/Components/Components/src/Forms/InputComponents/InputNumber.cs

```diff
@@ -100,98 +78,23 @@ protected override string FormatValueAsString(T value)
                     return null;
 
                 case int @int:
-                    return @int.ToString(CultureInfo.InvariantCulture);
+                    return BindConverter.FormatValue(@int, CultureInfo.InvariantCulture);
 
                 case long @long:
-                    return @long.ToString(CultureInfo.InvariantCulture);
+                    return BindConverter.FormatValue(@long, CultureInfo.InvariantCulture);
 
                 case float @float:
-                    return @float.ToString(CultureInfo.InvariantCulture);
+                    return BindConverter.FormatValue(@float, CultureInfo.InvariantCulture);
 
                 case double @double:
-                    return @double.ToString(CultureInfo.InvariantCulture);
+                    return BindConverter.FormatValue(@double, CultureInfo.InvariantCulture);
 
                 case decimal @decimal:
-                    return @decimal.ToString(CultureInfo.InvariantCulture);
+                    return BindConverter.FormatValue(@decimal, CultureInfo.InvariantCulture);
 
                 default:
                     throw new InvalidOperationException($"Unsupported type {value.GetType()}");
             }
         }
-
-        static bool TryParseInt(string value, out T result)
-        {
-            var success = int.TryParse(value, NumberStyles.Integer, CultureInfo.InvariantCulture, out var parsedValue);
-            if (success)
-            {
-                result = (T)(object)parsedValue;
-                return true;
-            }
-            else
-            {
-                result = default;
-                return false;
-            }
-        }
-
-        static bool TryParseLong(string value, out T result)
-        {
-            var success = long.TryParse(value, NumberStyles.Integer, CultureInfo.InvariantCulture, out var parsedValue);
-            if (success)
-            {
-                result = (T)(object)parsedValue;
-                return true;
-            }
-            else
-            {
-                result = default;
-                return false;
-            }
-        }
-
-        static bool TryParseFloat(string value, out T result)
-        {
-            var success = float.TryParse(value, NumberStyles.Float, CultureInfo.InvariantCulture, out var parsedValue);
-            if (success && !float.IsInfinity(parsedValue))
```

Comment:
Do the E2E tests still pass after this PR? If so it would be good to understand how. The `float.IsInfinity` check (and equivalent  for double below) was needed because we can't round-trip very large values through string representations.

For example, if the user enters into an `InputNumber` the string `123.456e100`, then `float.TryParse` treats that as "success" and returns the value "infinity". If this is then formatted as a string, it produces the string `∞`. Then when we try to write that back to the `InputNumber`, we can't because browsers don't accept it as a valid number. So ultimately to avoid exceptions with large values, we have to treats strings as unparseable if they don't give us a value that could later be written back to the client.

This is covered in the E2E test `InputNumberInteractsWithEditContext_NullableFloat`. If that's not failing without the `float.IsInfinity` check then I'd love to know why!

Reply:
When I try this out, what I'm seeing is that the value `123.456e100` does not successfully parse (returns false). I don't think anything has changed here in the logic of the test or conversion logic that would impact it.

I'm going to go ahead and add the check to treat infinity and nan as invalid, because I don't think we need them. Any disagreement?